homeassistant/components/ecovacs/vacuum.py

```diff
@@ -56,14 +63,13 @@ def __init__(self, device: sucks.VacBot) -> None:
         """Initialize the Ecovacs Vacuum."""
         self.device = device
         self.device.connect_and_wait_until_ready()
-        if self.device.vacuum.get("nick") is not None:
-            self._attr_name = str(self.device.vacuum["nick"])
-        else:
-            # In case there is no nickname defined, use the device id
-            self._attr_name = str(format(self.device.vacuum["did"]))
+        vacuum = self.device.vacuum
+
+        self.error = None
+        self._attr_unique_id = vacuum["did"]
+        self._attr_name = vacuum.get("nick", vacuum["did"])
 
-        self._error = None
-        _LOGGER.debug("Vacuum initialized: %s", self.name)
+        _LOGGER.debug("StateVacuum initialized: %s", self.name)
```

Comment:
Side note: We don't want side effects like logging in init methods. I'd remove this log.

Reply:
#96200